### PR TITLE
Update TestHelmNamespaceRepo to make it works in shared test environment (#75)

### PR DESCRIPTION
**Pull request by nan-yu**
Change the RepoSync to sync from original git source at the end of test to make test works in the shared test environment.

### CODE REVIEW

**Review by xinnywinne — APPROVED** [review]
/lgtm

File: e2e/testcases/helm_sync_test.go [hunk 1]
```diff
@@ -106,14 +104,16 @@ func TestHelmNamespaceRepo(t *testing.T) {
...
 	nt.RootRepos[configsync.RootSyncName].Add(nomostest.StructuredNSPath(repoSyncNN.Namespace, repoSyncNN.Name), rs)
 	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Update RepoSync to sync from a private Helm Chart without cluster scoped resources")
 	nt.WaitForRepoSyncs(nomostest.WithRepoSha1Func(helmChartVersion(privateNSHelmChart+":"+privateNSHelmChartVersion)), nomostest.WithSyncDirectoryMap(map[types.NamespacedName]string{repoSyncNN: privateNSHelmChart}))
 	if err := nt.Validate(rs.Spec.Helm.ReleaseName+"-"+privateNSHelmChart, testNs, &appsv1.Deployment{}); err != nil {
 		nt.T.Error(err)
 	}
+	// Change the RepoSync to sync from the original git source to make test works in the shared test environment.
+	rs.Spec.SourceType = string(v1beta1.GitSource)
+	nt.RootRepos[configsync.RootSyncName].Add(nomostest.StructuredNSPath(repoSyncNN.Namespace, repoSyncNN.Name), rs)
+	nt.RootRepos[configsync.RootSyncName].CommitAndPush("Update RepoSync to sync from the original git repository")
```
Comment (tiffanny29631) [inline e2e/testcases/helm_sync_test.go:116]: WaitForRepoSyncs for this step?

Comment (tiffanny29631) [reply e2e/testcases/helm_sync_test.go:116]: Would resetting .Spec.Helm to nil be safer

**Review by tiffanny29631 — COMMENTED** [review]
/lgtm

File: e2e/testcases/helm_sync_test.go (hunk 1 above)
Comment (sdowell) [reply e2e/testcases/helm_sync_test.go:116]: This is a cherry pick - we shouldn't alter the diff if possible

Comment (nan-yu) [reply e2e/testcases/helm_sync_test.go:116]: This is the cherry-pick PR and we would like it to be consistent with the commit on the main branch.

> WaitForRepoSyncs for this step?

Xin has moved this snippet to a cleanup block in #93. It is not necessary to wait for repo syncs to be synced as the test infra will validate that after it reset everything.

> Would resetting .Spec.Helm to nil be safer

Not necessary. The `.spec.helm` field will be ignored when `.spec.sourceType` is not `helm`.

Comment (tiffanny29631) [reply e2e/testcases/helm_sync_test.go:116]: Ack thanks for the explanation

**Review by tiffanny29631 — APPROVED** [review]
/lgtm

**Pull request merged**